<?xml version="1.0"?>
<!--XML-->
<launch>

    <!-- SETUP ARGUMENTS -->
    <arg name="gui" default="false"/>
    <arg name="camera_namespace" default="i3dr_stereo_camera"/>

    <arg name="stereo_algorithm" default="0" />
    <arg name="min_disparity" default="128" />
    <arg name="disparity_range" default="256" />
    <arg name="correlation_window_size" default="19" />
    <arg name="uniqueness_ratio" default="2" />
    <arg name="speckle_size" default="1000" />

    <!-- STEREO <MATCHER> -->
    <node ns="$(arg camera_namespace)" pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc" output="screen">
        <!-- Matcher Parameters -->
        <!-- see link below for details on matcher parameters -->
        <!-- http://wiki.ros.org/stereo_image_proc/Tutorials/ChoosingGoodStereoParameters -->
        <!-- algorithm 1: OpenCV block -->
        <!-- algorithm 2: OpenCV SGBM -->
        <param name="stereo_algorithm" value="$(arg stereo_algorithm)" />
        <param name="min_disparity" value="$(arg min_disparity)" />
        <param name="disparity_range" value="$(arg disparity_range)" />
        <param name="correlation_window_size" value="$(arg correlation_window_size)" />
        <param name="uniqueness_ratio" value="$(arg uniqueness_ratio)" />
        <param name="speckle_size" value="$(arg speckle_size)" />

        <!-- approximate sync used when cameras are not perfectly triggered in unison -->
        <param name="approximate_sync" value="true" />
        <param name="queue_size" value="50"/>
    </node>

    <!-- DISPLAY -->
    <!-- display phobos disparity image -->
    <node if="$(arg gui)" pkg="image_view" type="disparity_view" name="disparity_view" output="screen">
        <remap from="image" to="/$(arg camera_namespace)/disparity" />
    </node>

    <!-- run rqt control gui for adjusting stereo configuration -->
    <node if="$(arg gui)" pkg="rqt_reconfigure" type="rqt_reconfigure" name="display_controls" />

</launch>
